tests/errors/e_cython_parallel.pyx

```diff
@@ -149,6 +149,26 @@ with nogil, cython.parallel.parallel():
     with cython.parallel.parallel():
         pass
 
+cdef bint gil_function():
+    return True
+
+for i in prange(10, nogil=True, use_threads_if=gil_function()):
+    pass
+
+with nogil, parallel.parallel(use_threads_if=gil_function()):
+    pass
+
+def bar():
+
+    python_var = object()
+
+    cdef int i
+
+    for i in prange(10, nogil=True, use_threads_if=python_var):
+        pass
```

Comment:
This looks like a case that we should support. Not necessarily right now from the start, but it seems unexpected that the Python condition is not being evaluated as part of the `prange()` function call but only after releasing the GIL (which is simply triggered by another argument to the same function call). The order in which we release the GIL and evaluate the condition should be up to us and not get in the way of the users.

Reply:
> This looks like a case that we should support. 

I agree. I did it in this way because this is the way how `prange()` works currently and it is simple to implement this way.